### DISCUSSION TITLE
Add ExternalTexture to Godot 4

### DISCUSSION THREAD

**Discussion by vvwx** [godotengine/godot-proposals#4902] [Engine Core]
Recently I've noticed that ExternalTexture is absent from Godot 4, and I cannot find anything related to it under [issues](https://github.com/godotengine/godot/issues?q=is%3Aissue+is%3Aopen+ExternalTexture), [proposals](https://github.com/godotengine/godot-proposals/issues?q=is%3Aissue+is%3Aopen+ExternalTexture), and [discussions](https://github.com/godotengine/godot-proposals/discussions?discussions_q=ExternalTexture).

It looks like the project converter from Godot 3 -> Godot 4 maps ExternalTexture to ImageTexture ([seen here](https://github.com/godotengine/godot/blob/1e553e34fbc74e16fbecc59406c97e3a0f4a5698/editor/project_converter_3_to_4.cpp#L1290)), however it looks like [ImageTexture](https://github.com/godotengine/godot/blob/master/scene/resources/texture.cpp#L130-L338) doesn't do what [ExternalTexture](https://github.com/godotengine/godot/blob/3.x/scene/resources/texture.cpp#L2859-L2916) does in 3.x (granted a lot has changed so maybe I'm missing something). 

I was considering submitting this as a bug with the `regression` tag, but figured I'd start this as a discussion first.

Here is the original [PR](https://github.com/godotengine/godot/pull/35306)

**Comment by xix-xeaon** [comment]
According to the [docs](https://docs.godotengine.org/en/stable/classes/class_externaltexture.html) in Godot 3, ExternalTexture is based on an extension to OpenGL ES and only supported on Android. Since Godot 4 is all about Vulcan that's probably why it was removed. Of course it might eventually get put back in since @clayjohn is working on adding back GLES3 but I don't know, it's probably not a priority.

**Comment by queezle42** [comment]
The feature should, in theory, be usable on Linux. The related OpenGL ES extensions (`GL_OES_EGL_image_external` for GLES2 and `GL_OES_EGL_image_external_essl3` for GLES3) are available on Linux and I am using them to intercept- and modify Wayland windows.

If the feature gets added to Godot 4 I'd like to use it to render desktop applications inside of Godot.
